test/common/common/log_macros_test.cc

```diff
@@ -118,6 +123,80 @@ TEST(Logger, checkLoggerLevel) {
   EXPECT_THAT(test_obj.executeAtTraceLevel(), testing::Eq(2));
 }
 
+void spamCall(std::function<void()>&& call_to_spam, const uint32_t num_threads) {
+  std::vector<std::thread> threads(num_threads);
+  std::promise<void> signal_all_threads_running;
+  std::shared_future<void> future(signal_all_threads_running.get_future());
+
+  for (auto& thread : threads) {
+    thread = std::thread([future, &call_to_spam] {
+      future.wait();
+      call_to_spam();
+    });
+  }
+  // Allow threads to accrue on future.wait() to maximize concurrency on the call
+  // we are testing.
+  sleep(1);
```

Comment:
I think we want to avoid relying on real-time for tests. @jmarantz do you see a way to cleanly add an injected time source for the periodic macro that @oschaaf introduced in this PR? I'm thinking maybe we need a time source singleton for logging and do singleton injection, to avoid having to worry about time source at call site.

Reply:
In 371fa5dffe018d2a594008a07d242071e6e1424d I rewrote the test to not rely on `sleep()`. 
Also backed out `ENVOY_LOG_PERIODIC` with the intent to do a follow up for adding that after this. 
(so we can take a look and discuss that separately)